Esri/arcgis-maps-sdk-swift-samples
# [Update] Add Mac Catalyst support

## Description

This PR adds Mac Catalyst as a deployment destination to the app. It also includes the following changes

- Remove "Mac Designed for iPad" as Mac Catalyst is a better experience, plus also makes it possible to run on Intel Macs.
- Remove toolkit from "Embed Frameworks" because it is a (local) source Swift package, and will produce error if embedded.
- Moved the README files to "[UNLOCALIZED_RESOURCES_FOLDER_PATH](https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html#//apple_ref/doc/uid/TP40003931-CH3-SW82)/READMEs", aka "Samples.app/Contents/Resources/READMEs". Semantically speaking, it should have been here already, as all the bundle resources should go under the `Resources` subpath, instead of under the root of the bundle.
- Namespace collision for `ArcGIS.Polygon` is resolved.
- Minimum deployment targets are added. Not sure what they are for, but I saw [toolkit also added them](https://github.com/ArcGIS/arcgis-maps-sdk-swift-toolkit/pull/201/files#diff-ad8ed6ec9639f261bb24472a1258373973c29184b843c51c69859078472dd7e1R380).
![targets](https://user-images.githubusercontent.com/9660181/205744501-76f8d378-d355-4b05-bcc1-e7e4b3634d8d.png)

## Linked Issue(s)

- `swift/issues/3153`

## How To Test

- Run on iPhone, iPad Simulator
- Run on Mac Catalyst
- Everything should work as before, except that on-demand resource is not working on Mac Catalyst (as expected)

## Screenshots

![Mac-Catalyst](https://user-images.githubusercontent.com/9660181/205744785-87ee2df0-bf70-422d-b3f0-63ef14971e8b.png)

## To Discuss

- (Slack) Team ID should be removed?
